procrustes/permutation.py

```diff
@@ -77,57 +96,42 @@ def permutation(
+        \atop \sum_{i=1}^n [\mathbf{P}]_{ij} = \sum_{j=1}^n [\mathbf{P}]_{ij} = 1} \right. \right\}}
+            \text{Tr}\left[\mathbf{P}^\dagger\mathbf{A}^\dagger\mathbf{B} \right]
 
-    Here, :math:`\mathbf{P}_{n \times n}` is the permutation matrix. The solution is to relax the
-    problem into a linear programming problem and note that the solution to a linear programming
-    problem is always at the boundary of the allowed region, which means that the solution can
-    always be written as a permutation matrix,
+    The solution is found by relaxing the problem into a linear programming problem. The solution
+    to a linear programming problem is always at the boundary of the allowed region. So,
 
     .. math::
-       \underbrace{\text{max}}_{\left\{\mathbf{P} \left| {p_{ij} \in \{0, 1\}
-                   \atop \sum_{i=1}^n p_{ij} = \sum_{j=1}^n p_{ij} = 1} \right. \right\}}
+       \underbrace{\text{max}}_{\left\{\mathbf{P} \left| {[\mathbf{P}]_{ij} \in \{0, 1\}
+       \atop \sum_{i=1}^n [\mathbf{P}]_{ij} = \sum_{j=1}^n [\mathbf{P}]_{ij} = 1} \right. \right\}}
           \text{Tr}\left[\mathbf{P}^\dagger\mathbf{A}^\dagger\mathbf{B} \right] =
-       \underbrace{\text{max}}_{\left\{\mathbf{P} \left| {p_{ij} \geq 0
-                   \atop \sum_{i=1}^n p_{ij} = \sum_{j=1}^n p_{ij} = 1} \right. \right\}}
+       \underbrace{\text{max}}_{\left\{\mathbf{P} \left| {[\mathbf{P}]_{ij} \geq 0
+       \atop \sum_{i=1}^n [\mathbf{P}]_{ij} = \sum_{j=1}^n [\mathbf{P}]_{ij} = 1} \right. \right\}}
           \text{Tr}\left[\mathbf{P}^\dagger\left(\mathbf{A}^\dagger\mathbf{B}\right) \right]
 
-    This is a matching problem and can be solved by the Hungarian method. Note that if
-    :math:`\mathbf{A}` and :math:`\mathbf{B}` have different numbers of items, you choose
-    the larger matrix as :math:`\mathbf{B}` and then pad :math:`\mathbf{A}` with rows/columns
-    of zeros.
+    This is a matching problem and can be solved by the Hungarian method.
 
     """
     # check inputs
     new_a, new_b = setup_input_arrays(
-        a,
-        b,
-        remove_zero_col,
-        remove_zero_row,
-        pad,
-        translate,
-        scale,
-        check_finite,
-        weight,
+        a, b, unpad_col, unpad_row, pad, translate, scale, check_finite, weight,
     )
+    # if number of rows is less than column, the arrays are made square
+    if (new_a.shape[0] < new_a.shape[1]) or (new_b.shape[0] < new_b.shape[1]):
+        new_a, new_b = _zero_padding(new_a, new_b, "square")
+
     # compute permutation Procrustes matrix
     array_p = np.dot(new_a.T, new_b)
-    array_c = np.full(array_p.shape, np.max(array_p))
-    array_c -= array_p
+    array_c = np.full(array_p.shape, np.max(array_p)) - array_p
     array_u = np.zeros(array_p.shape)
     # set elements to 1 according to Hungarian algorithm (linear_sum_assignment)
     array_u[linear_sum_assignment(array_c)] = 1
```

Comment:
Reading the docstring and comments (as well as the paper), it was not clear to me why the cost function is defined this way, so I think we need to add some clarification. @fwmeng88  can you please share the reference/notes used? Thanks.

Reply:
The Hungarian assignment seeks for minimizing the assignment cost, `x=A<sup>T</sup>B`. But in our scenario, we want to maximize the cost, which is equivalently to maximize the overlap (permutation Procrustes). Therefore, we can say we want to minimize `-x`. But in some cases, there can be a negative value, therefore, we add all the elements in the cost function with the maximum value to ensure all the elements are positive. 

I am not sure if I explain it clear or not. Maybe @PaulWAyers have some ideas.
@FarnazH

Reply:
This is a lot clearer. Can we be sure the documentation explains this?

@fwmeng88 was this something you cooked up, or is there a reference? The reference I have doesn't mention the negative and shift, though it is kinda obvious we should still give credit if we took it from somewhere.

Reply:
I didn't remember we referred some references for this problem settings. But when I checked up this morning, there is an old paper that mentioned this and we should cite this, [Variants of the Hungarian method for assignment problems](https://onlinelibrary.wiley.com/doi/abs/10.1002/nav.3800030404). The manuscript has been updated accordingly.

Reply:
This code snippet shows a draft docstring that can be added for single-sided permutation Procrustes.

```reStructuredText
    This is a matching problem and can be solved by the Hungarian method. The Hungarian assignment
    algorithm seeks to minimize the assignment cost,
    :math:`\mathbf{C} = \mathbf{A}^T \mathbf{B}`. But a maximum permutation sum is
    required for permutation Procrustes where we used `-\mathbf{C} = \mathbf{A}^T \mathbf{B}` as our
    objective function. A second strategy we used here is to ensure the non-negativeness by adding
    the maximum element to the objective matrix.
```

I don't think it a wise choice to push commits to your branch, which can help keep the history clean. Hope it helps a little.
@FarnazH

Reply:
Thanks @fwmeng88. I used C=A.T B as the cost function, as the `scipy.optimize.linear_sum_assignment` had an argument for maximization, so I set `maximize=True`.